hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileBlock.java

```diff
@@ -1742,6 +1760,13 @@ protected HFileBlock readBlockDataInternal(FSDataInputStream is, long offset,
         onDiskSizeWithHeader = getOnDiskSizeWithHeader(headerBuf, checksumSupport);
       }
 
+      // if checksumType on headerBuf validate fail,the headerBuf may corrupted
```

Comment:
This comment should be a more clear, like the comment above the next check. I propose something like: "Inspect the header's checksumType for known valid values. If we don't find such a value, assume that the bytes read are corrupted and act accordingly."